crates/typst/src/visualize/color.rs

```diff
@@ -1267,6 +1308,30 @@ impl Color {
         self
     }
 
+    /// Scales the alpha value of a color by a given amount.
+    ///
+    /// For positive scales, computes `alpha + scale - alpha * scale`.
+    /// For non-positive scales, computes `alpha + alpha * scale`.
+    fn scale_alpha(self, scale: f32) -> StrResult<Color> {
```

Comment:
```suggestion
    fn scale_alpha(self, scale: Ratio) -> StrResult<Color> {
```
should we leave it as a ratio until here?